conda/plugins/manager.py

```diff
@@ -168,6 +168,13 @@ def get_hook_results(self, name: str) -> list:
             )
         return plugins
 
+    def get_solvers(self) -> dict[str, type[Solver]]:
+        """Return a mapping from solver name to solver class."""
+        return {
+            solver.name.lower(): solver.backend
+            for solver in self.get_hook_results("solvers")
+        }
```

Comment:
The pattern I thought those getters were supposed to have didn't include deconstructing the domain object, but return a mapping between the name of the plugin instance and the domain object

```suggestion
        return {
            solver_hook.name.lower(): solver
            for solver in self.get_hook_results("solvers")
        }
```

Reply:
see https://github.com/conda/conda/pull/13057/commits/86497e13084e52de7fcc7790fbd26345e7a3e1e2